tests/cupy_tests/core_tests/test_raw.py

```diff
@@ -714,6 +725,19 @@ def test_template_failure(self):
                            match='named symbol not found'):
             mod.get_function('my_sqrt<double>')
 
+    def test_raw_pointer(self):
+        mod = cupy.RawModule(code=test_cast, backend=self.backend)
```

Comment:
I think you are not ensuring that `MemoryPointer` is being used.

Reply:
What do you mean? `arr` here is a `MemoryPointer` instance. Perhaps I should rename it?

Reply:
Sorry I was confused for a bit and thought you were directly passing the array to the kernel instead of the pointer.
LGTM